redigo.go

```diff
@@ -44,51 +46,56 @@ func (b *Bundle) Build(builder *di.Builder) error {
 	return builder.Add(di.Def{
 		Name: BundleName,
 		Build: func(ctn di.Container) (_ interface{}, err error) {
-			var cfg *viper.Viper
-			if err = ctn.Fill(viper.BundleName, &cfg); err != nil {
+			var v *viper.Viper
+			if err = ctn.Fill(viper.BundleName, &v); err != nil {
 				return nil, err
 			}
 
-			var conf redisConf
-			if err = cfg.UnmarshalKey("redis", &conf); err != nil {
-				return nil, err
+			// set default
+			v.SetDefault("redis.port", "6379")
+			v.SetDefault("redis.max_idle", 3)
+			v.SetDefault("redis.idle_timeout", 240*time.Second)
+
+			var cfg = config{
+				Host:        v.GetString("redis.host"),
+				Port:        v.GetString("redis.port"),
+				Password:    v.GetString("redis.password"),
+				MaxIdle:     v.GetInt("redis.max_idle"),
+				MaxActive:   v.GetInt("redis.max_active"),
+				IdleTimeout: v.GetDuration("redis.idle_timeout"),
 			}
 
-			if conf.MaxIdle < 0 {
-				return nil, errors.New("redis.MaxIdle should be greater then 0")
+			// validating
+			if cfg.Host == "" {
+				return nil, errors.New("redis.host should be set")
 			}
 
-			// set default
-			if conf.MaxIdle == 0 {
-				conf.MaxIdle = 3
+			if cfg.MaxIdle < 0 {
+				return nil, errors.New("redis.MaxIdle should be greater then 0")
 			}
 
-			if conf.MaxActive < 0 {
+			if cfg.MaxActive < 0 {
 				return nil, errors.New("redis.MaxActive should be greater or equal to 0")
```

Comment:
Invalid name of config key, should be redis.max_active

Reply:
fixed